### ISSUE TITLE
Screen off functionality

### ISSUE THREAD

**Issue by L0czki** [Maassoft/ColorControl#203]
Hey,

was wondering if it's possilble to add screen off functionality. Power off is nice, but I guess screen off is a bit faster response.

Cheers,

**Comment by GTMoraes** [comment]
Just screen off, without turning the display off?

You can use the Screen Off feature on Windows's Power Settings

**Comment by L0czki** [comment]
Yeah I power it off when screensaver kicks in- that's easy. That's not what I mean. This is LG functionality (but I know it's also in Sony TV), that you turn display off, but the TV is still on. Something like black screen saver - but from hardware perspective I suppose.

https://youtu.be/X8De-T4aTgM?t=35 you can see it here in this vid.

**Comment by GTMoraes** [comment]
Ah, I understood it, now.

Yeah, that's to be evaluated by the dev if it's a triggerable option.

In any case, technically, running a black screen essentially turns the OLED panel off, as "black" for OLEDs means OFF.           
You can use the "Blank" screensaver for that purpose. 

If I were to guess, I think that built-in feature is exactly that: just displaying a black image over the display, to "turn it off"

**Comment by Maassoft** [comment]
Currently it is not easy to to trigger a preset whenever the screensaver starts, but I think it can be done using the "Process switch" trigger, filling in "scrnsave.scr" at the "Included processes" and setting up steps like this: energySaving(screen_off)
But then you'll have to make another "default" preset which gets activated in all other cases and disables the energy saving.
So that's probably a bit of a hassle, maybe I'll add more trigger options in a new version.

By the way, there was a user requesting something like this before, but found out that running the simple "blank screensaver" from Windows has about the same effect as the the "Energy Saving - Screen Off" setting on the TV. The power usage is virtually the same, so there would be no gain.

**Comment by ravensergio** [comment]
I was about to suggest the screen off feature, and it could be implemented maybe in Your already S button ( where the delay for the ScreenSaver is. 
Something like: 
- Use Screen Off for "powerOff when Screensaver activates" instead of Power Off.
- Use Screen ON for "powerOn when Screensaver deactivates" - so 2 more checkboxes or something in that "S" form.

Btw, I tried the Process Switch for screen Off, but here Screen Off doesn't work even if I manually set  the " Energy Saving-screen_off" - LG C2. It does nothing for me. Everything else seems to work, so is not the connection I'm positive.

> running the simple "blank screensaver" from Windows has about the same effect as the the "Energy Saving - Screen Off" 

That is true somehow, but I have read that is not exactly the same, since the leds still run power through them, and if you have raised the black level by a bit (not perfect black in blank screensaver) the leds are NOT turning off for sure, so screen Off would be the guaranteed way to go.

Also, Screen Off is preferable over Power OFF since screensaver activates for me in 3-4 minutes ( cause OLED reasons), and Win 10/11 is wierd sometimes and detects no monitor and messes up my resolution, and when I turn the TV back on, icons and windows location/sizes gets messed up. Not always, but it happens here.

For now I resolved this screen off thing via Ahk and HomeAssistant api ( for executing screen off/on) by checking for example the "Mystify.scr" process, and reverting if the process is no more.

**Comment by ravensergio** [comment]
> Is there a way I can trigger the TV's "Screen Off" function via keyboard shortcut? I'd like to press a shortcut on my keyboard and start the TV's screen off feature, so I don't have to use the remote( using the remote it's; Settings cog, up, up, up, OK). I have an LG C2.

You could in theory. For me the command is not working ( unless screen_off is not the one we need):
Make a new profile thing in LG Controller tab like this:

![image](https://user-images.githubusercontent.com/32698066/233859366-a7f4e7cc-6885-4063-a985-9cfad09c593a.png)

You may need to make a separate profile/shortcut for Screen On though.

**Comment by vlatko27** [comment]
> > Is there a way I can trigger the TV's "Screen Off" function via keyboard shortcut? I'd like to press a shortcut on my keyboard and start the TV's screen off feature, so I don't have to use the remote( using the remote it's; Settings cog, up, up, up, OK). I have an LG C2.
> 
> You could in theory. For me the command is not working ( unless screen_off is not the one we need): Make a new profile thing in LG Controller tab like this:
> 
> ![image](https://user-images.githubusercontent.com/32698066/233859366-a7f4e7cc-6885-4063-a985-9cfad09c593a.png)
> 
> You may need to make a separate profile/shortcut for Screen On though.

Hmm..that does not seem to do anything?

**Comment by ravensergio** [comment]
It doesn't seem to work for me either, as I said, idk if is not this one or needs its implementation re-checked by @Maassoft . 
I thought it is just me that has this issue. I could not find another command that should trigger  Screen off besides - energySaving -> ScreenOff.
If it is the case and this is not working as it should now, then we have to wait for an update or some help from the author maybe.

**Comment by ravensergio** [comment]
Whan testing switch the tab from 'LG controller' go to game launcher or options tab first then try the shortcuts.
Here is working properly, I've set up the same shortcut as You, screen turned off.

**Comment by vlatko27** [comment]
> When testing switch the tab from 'LG controller' go to game launcher or options tab first then try the shortcuts. Here is working properly, I've set up the same shortcut as You, screen turned off.

Heureka, it works! Thanks guys!

**Comment by Maassoft** [comment]
v9.2.0.0 now has the options accessible via the 'S'-button to turn the screen off instead of powering off the TV.

**Comment by Maassoft** [comment]
Yes, I get what you mean. I was already pondering about an new setting: an extra timeout that gets activated after the screen is turned off and will power off the TV after X minutes. After the TV's screen has been turned off, you have the option to power off the TV after a configurable time. That's should get most cases covered.

**Comment by ravensergio** [comment]
If ScreenOff instead of power OFF triggers on the screensaver only, then other power off functions could be kept in place.
For example, I would have screen off after 3-4 minutes, but if I am not coming back in 1 hr I've set my pc to sleep so the power off on standby or shutdown etc should trigger. But in other cases, if someone doesn't put the pc to sleep etc, they can choose to 'Power OFF' on screensaver like before.
Ofc the screenOff is not for prolonged time, I like it because my pc is set to shutdown if idle after 1 hr anyways, so ScreenOff is beneficial to me, faster to wake up after screensaver OFF, my windows don't get messed up (because tv shuts down hdmi when turned off and Windows goes crazy with the resolutions/win positions etc). Recently I learned that if I keep my TV in Always ready mode, the HDMI signal wont get interrupted even on tv power off , but I did not tested. The ScreenOFF works better in my case.
Thank You for the update, this is awesome!
